bears/general/FilenameBear.py

```diff
@@ -74,12 +82,38 @@ def run(self, filename, file,
         if ignore_uppercase_filenames and filename_without_extension.isupper():
             return
 
-        if new_name != filename_without_extension:
-            diff = Diff(file, rename=os.path.join(head, new_name + extension))
+        messages = []
 
-            yield Result(
-                self,
+        if new_name != filename_without_extension:
+            messages.append(
                 'Filename does not follow {} naming-convention.'.format(
-                    file_naming_convention),
-                diff.affected_code(filename),
-                diffs={filename: diff})
+                    file_naming_convention))
+
+        if not filename_without_extension.startswith(filename_prefix):
+            new_name = filename_prefix + new_name
+            messages.append(
+                'Filename does not use the {} prefix.'.format(
+                    filename_prefix))
+
+        if not filename_without_extension.endswith(filename_suffix):
+            new_name = new_name + filename_suffix
+            messages.append(
+                'Filename does not use the {} suffix.'.format(
+                    filename_suffix))
+
+        if messages:
+            diff = Diff(file, rename=os.path.join(
+                head, new_name + extension))
+
+            if len(messages) > 1:
+                yield Result(
+                    self,
+                    '\n'.join('- ' + message for message in messages),
+                    diff.affected_code(filename),
+                    diffs={filename: diff})
+            else:
+                yield Result(
+                    self,
+                    messages[0],
+                    diff.affected_code(filename),
+                    diffs={filename: diff})
```

Comment:
don't generate 2 results, but two different messages :D
```
if len(messages) > 1:
    result_message = '\n'.join('- ' + message for message in messages)
else:
    result_message = messages[0]

yield Result(
    self, result_message, diff.affected_code(filename), diffs={filename:diff})
```
or you can compact the `if`-statement to an `if`-expression if you like:
```
result_message = ('\n'.join('- ' + message for message in messages)
                  if len(messages) > 1 else
                  messages[0])
```